### ISSUE TITLE
Does not build with coq-flocq 4.0.0

### ISSUE THREAD

**Issue by intoverflow** [AbsInt/CompCert#423]
Build failed for me at this line: https://github.com/AbsInt/CompCert/blob/v3.9/lib/IEEE754_extra.v#L35

`Z` is not defined. Maybe flocq changed their Exports?

**Comment by xavierleroy** [comment]
I was not aware that Flocq 4.0 was released, but I'm not surprised it breaks CompCert.  See #368 for a previous step towards Flocq 4.0 support.  I'll let @silene update #368 if/when he sees fit.

In the meantime, please don't use Flocq 4.0.0 with CompCert, or use the "vendored" version of Flocq that is included in the CompCert distribution and that is guaranteed to work.

**Comment by silene** [comment]
> Maybe flocq changed their Exports?

Yes, Flocq was a bit too eager to export files from other libraries. So, the major release was a good time to clean things a bit. It is easy to make CompCert forward-compatible on that point. But there are some other changes that I have little hope to accommodate. So, at some point, we need to make the switch to Flocq 4.

> I'll let @silene update #368 if/when he sees fit.

There is a larger issue at hand, which is that Flocq 4 requires Coq 8.12. So, that would implicitly bump the minimal requirement of CompCert from 8.8 to 8.12. Thus, I am not quite sure what to do.

A potential solution would be that I frankenstein the vendored version of Flocq 3, so that it provides an interface somehow compatible with Flocq 4 (as far as CompCert is concerned), yet compiles with Coq 8.8. In other words, if a user wants to use an external Flocq, they would need Coq 8.12 at least. But the vendored one would still compile with older releases of Coq.

**Comment by xavierleroy** [comment]
> So, that would implicitly bump the minimal requirement of CompCert from 8.8 to 8.12. Thus, I am not quite sure what to do.

I like it when CompCert can work with several major versions of Coq (just in case a terrible bug was introduced in the latest version), but I don't think we absolutely need to go back all the way to Coq 8.8.  I'd like to get a bit more confidence with Coq 8.14, but after that I'm OK with requiring Coq 8.12 at least.

**Comment by xavierleroy** [comment]
Also: please don't "frankenstein" anything just for CompCert's sake.  The purpose of the vendored version of Flocq is to follow the development of Flocq, but possibly with a delay as incompatible changes in Flocq are being addressed.

**Comment by xavierleroy** [comment]
For the record: the switch of CompCert to Flocq 4 is scheduled for the next public release, in May or June 2022.
